bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/IDFBuildConfiguration.java

```diff
@@ -1245,6 +1245,10 @@ private IFile getFileForCMakePath(String sourceFileName, IProject project)
 				pathtolookfor = getIdfToolsPath();
 				startIndex = sourceFile.indexOf(pathtolookfor);
 			}
+			if (startIndex == -1) // source file still not found means it was part of another esp-idf
+			{
+				return null;
+			}
```

Comment:
That means`pathtolookfor = getIdfToolsPath();` was not returning the correct IDF_PATH, I wonder why IDF_PATH is not configured correctly and what are the other issues it might cause!

If user switched to a different esp-idf version then IDF_PATH would have modified accordingly.

Reply:
Hi @kolipakakondal, the IDF_PATH from getIdfToolsPath() is correct, the issue happens because sourceFile is pointing to the different esp-idf I think sourceFile is provided based on the files from the build folder. This issue happens when we build the project with the different esp-idf version, then switch esp-idf and trying to build project again without cleaning build folder